jetty-util/src/main/java/org/eclipse/jetty/util/component/Dumpable.java

```diff
@@ -155,41 +156,51 @@ static void dumpObjects(Appendable out, String indent, Object object, Object...
 
         if (object instanceof Container)
         {
-            dumpContainer(out, indent, (Container)object, extras == 0);
+            dumpContainer(out, indent, (Container)object, list);
         }
         if (object instanceof Iterable)
         {
-            dumpIterable(out, indent, (Iterable<?>)object, extras == 0);
+            dumpIterable(out, indent, (Iterable<?>)object, list);
         }
         else if (object instanceof Map)
         {
-            dumpMapEntries(out, indent, (Map<?, ?>)object, extras == 0);
+            dumpMapEntries(out, indent, (Map<?, ?>)object, list);
         }
 
-        if (extras == 0)
+        if (list.size() == 0)
             return;
 
         int i = 0;
         for (Object item : extraChildren)
         {
             i++;
-            String nextIndent = indent + (i < extras ? "|  " : "   ");
+            String nextIndent = indent + (i < list.size() ? "|  " : "   ");
             out.append(indent).append("+> ");
             if (item instanceof Dumpable)
                 ((Dumpable)item).dump(out, nextIndent);
             else
                 dumpObjects(out, nextIndent, item);
         }
     }
-
+    
+    @Deprecated
     static void dumpContainer(Appendable out, String indent, Container object, boolean last) throws IOException
+    {
+        dumpContainer(out, indent, object, (last ? Collections.emptyList() : Collections.singletonList(new Object())));
```

Comment:
Why are we dumping this `new Object()` created on the fly?
At least it deserves a comment.

Reply:
We are actually _not_ dumping this object at all. If you look at the implementation of dumpContainer, it will _skip_ any bean children of the Container that match any of the objects passed in as the list of extras. In this case, we are passing in either the empty list, or a singleton object as extras, neither of which will ever match anything. I've done this simply to maintain for backwards compatibility the old method signature of dumpContainer(Appendable, String, Container, boolean) - the fact of the matter is that we do not have any code anymore that calls that old method signature, so this (now deprecated) method can be entirely removed in jetty-10.